subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/IsolatingIncrementalAnnotationProcessingIntegrationTest.groovy

```diff
@@ -109,13 +151,14 @@ class IsolatingIncrementalAnnotationProcessingIntegrationTest extends AbstractIn
         run "compileJava"
 
         then:
-        outputs.recompiledFiles("A", "AHelper", "AHelperResource.txt")
+        outputs.recompiledFiles("A", "AHelper", "AHelperResource.txt", "Dependent")
     }
 
     def "source file is recompiled when dependency of generated file changes"() {
         given:
         withProcessor(new OpaqueDependencyProcessor())
         java "@Thingy class A {}"
+        java "class Unrelated {}"
```

Comment:
This test wasn't actually testing incremental processing, because the processor was not marked as incremental. It just did a full recompile, so it would have passed even if we completely broke the relevant logic. Inserting this unrelated class here ensures that we really had an incremental compilation.